### PR TITLE
fix problem with "plugin tag are already registered" debug messages

### PR DESCRIPTION
**Pull request by naucon**
thank you for your hard work on this great bundle.

I noticed a lot of "plugin tag already registered" debug messages on a project of mine.
The project uses the Symfony AppCache and a lot of ESI's. With every ESI render() a block of this debug messages were logged.

```
[2016-03-18 18:43:33] SmartyBundle.DEBUG: SmartyException caught: Plugin tag "path" already registered.
[2016-03-18 18:43:33] SmartyBundle.DEBUG: SmartyException caught: Plugin tag "path" already registered.
[2016-03-18 18:43:33] SmartyBundle.DEBUG: SmartyException caught: Plugin tag "url" already registered.
[2016-03-18 18:43:33] SmartyBundle.DEBUG: SmartyException caught: Plugin tag "url" already registered.
```

The debug messages are caused because the method registerPlugins() is called on the method render(). registerPlugins() registers the plugins on the Smarty instance regardless if it is already registered.

My solution is to track which plugins are already registered and skip those that are being already registered. That allows to add plugins between two or more render() calls.

I would appreciate if you could merge my pull request.

### CODE REVIEW

File: Extension/Plugin/PluginInterface.php [hunk 1]
```diff
@@ -35,4 +35,14 @@
 interface PluginInterface
 {
     public function getType();
+
+    /**
+     * Get the plugin name.
+     */
+    public function getName();
```
Comment (noisebleed) [inline Extension/Plugin/PluginInterface.php]: By adding these 2 new methods to the contract we force existing Plugins to implement these methods. Meaning it will break current plugins in SmartyBundle and those created by SmartyBundle users in their projects. It's just not visible in tests because there is no coverage atm for plugins :(

File: SmartyEngine.php [hunk 2]
```diff
@@ -569,13 +570,21 @@ public function getPlugins($extensionName = false)
...
+    /**
+     * register plugin to Smarty.
+     *
+     * @return  void
+     */
+    protected function registerPlugin(PluginInterface $plugin)
+    {
+        // verify that plugin isn't registered yet. That would cause a SmartyException.
+        if (!in_array($plugin->getType() . '_' . $plugin->getName(), $this->registeredPlugins)) {
+            $this->smarty->registerPlugin($plugin->getType(), $plugin->getName(), $plugin->getCallback());
```
Comment (noisebleed) [inline SmartyEngine.php]: Would be preferable to "ask" Smarty if the plugin is already registered to avoid keeping state in two places leading to inconsistency potentially. Not sure if Smarty provides this though.

**Comment by noisebleed** [comment]
Thanks for your contribution @naucon. Added a comment to the PR but happy to merge as is.

File: SmartyEngine.php (hunk 2 above)
Comment (naucon) [inline SmartyEngine.php]: I totally agree.

Smarty stores the plugins in a public attribute `$smarty->registered_plugins[$type][$name]` therefore it is accessible. I send you another pull request.
